forc-pkg/src/lock.rs

```diff
@@ -136,7 +136,7 @@ impl PkgLock {
     ///
     /// If this package's name is not enough to disambiguate it from other packages within the
     /// graph, this returns `<name> <source>`. If it is, it simply returns the name.
-    pub fn name_disambiguated(&self, disambiguate: &HashSet<&str>) -> Cow<str> {
+    pub fn name_disambiguated(&self, disambiguate: &HashSet<&str>) -> Cow<'_, str> {
```

Comment:
This Clippy lint tries to remove ambiguity by setting the lifetime explicitly, but IMO just causes confusion and clutter in all of the cases seen in this PR. What is the added value of `'_` in this and other cases, especially in cases we have `<'_, '_> like in `SubstTypesContext<'_, '_>`? IMO no added value, just confusion.

Personally, I am for turning this lint off.

Others? @IGI-111 @zees-dev

Reply:
I am also for turning this off, if that is fine I will be going forward with it 👍  cc @JoshuaBatty

Reply:
fine with me.